src/qibolab/instruments/abstract.py

```diff
@@ -12,6 +12,14 @@
 class InstrumentSettings:
     """Container of settings that are dumped in the platform runcard yaml."""
 
+    @staticmethod
+    def dict_factory(x):
+        """Factory used for the dataclass ``asdict`` method.
+
+        Can be used to exclude fields from the dictionary.
+        """
+        return x
```

Comment:
?

Reply:
Stackoverflow way of omitting dataclass fields from `asdict`: https://stackoverflow.com/questions/61606700/python-dataclasses-omit-field-from-asdict. Here the default omits nothing, but we may decide to overwrite and omit something in the subclasses. For example in the `LocalOscillatorSettings` I omit `ref_osc_source` because I don't want it to appear to the yaml when we dump the platform.

Btw, thanks for checking the changes, just noting that I am also planning to add some tests using the dummy oscillator. Now that most functions are lifted we should be able to increase the coverage without access to the instruments.

Reply:
> Stackoverflow way of omitting dataclass fields from `asdict`: [stackoverflow.com/questions/61606700/python-dataclasses-omit-field-from-asdict](https://stackoverflow.com/questions/61606700/python-dataclasses-omit-field-from-asdict).

This is fine, it is even documented in stdlib:
https://docs.python.org/3/library/dataclasses.html#dataclasses.asdict

> Here the default omits nothing, but we may decide to overwrite and omit something in the subclasses.

I missed the connection with `serialize`, maybe cite the place where it is supposed to be used in the docstring (for developers), and declare that by default it does nothing (I did not understand in the beginning, but it is perfectly fine).

> Btw, thanks for checking the changes, just noting that I am also planning to add some tests using the dummy oscillator. Now that most functions are lifted we should be able to increase the coverage without access to the instruments.

Great to hear!